src/controllers/application-controller.ts

```diff
@@ -193,4 +211,81 @@ export class ApplicationController {
 			});
 		}
 	};
+
+	/**
+	 * GET /job-roles/:id/applicants
+	 * Renders the applicants list for a specific job role
+	 */
+	public getApplicants = async (req: Request, res: Response): Promise<void> => {
+		try {
+			const id = req.params["id"];
+			const jobRoleId = validateJobRoleId(id);
+
+			if (jobRoleId === null) {
+				res.status(400).render("error.njk", {
+					message:
+						"Invalid job role ID provided. Please provide a valid numeric ID.",
+				});
+				return;
+			}
+
+			// Get pagination parameters
+			const page = Number.parseInt(req.query["page"] as string, 10) || 1;
+			const limit = Number.parseInt(req.query["limit"] as string, 10) || 10;
+
+			// Validate pagination parameters
+			if (page < 1 || limit < 1 || limit > 50) {
```

Comment:
it would be good to assign 50 to some constant somewhere with a clear name.
I presume this is `MAX_PAGE_SIZE`?